R/restore_paths.R

```diff
@@ -0,0 +1,138 @@
+#' @include generics.R
+NULL
+
+#' @param pxl_files_dir The directory where the PXL files are stored
+#' @param verbose Print messages
+#' @param ... Additional arguments. Currently not used.
+#'
+#' @rdname RestorePaths
+#' @method RestorePaths MPXAssay
+#'
+#' @examples
+#' library(pixelatorR)
+#' library(dplyr)
+#'
+#' # Load example data as a Seurat object
+#' pxl_file <- system.file("extdata/five_cells",
+#'                         "five_cells.pxl",
+#'                         package = "pixelatorR")
+#'
+#' # Copy PXL file to tempdir
+#' tmp_pxl_file <- file.path(fs::path_temp(), "five_cells.pxl")
+#' fs::file_copy(pxl_file, tmp_pxl_file)
+#' seur_obj <- ReadMPX_Seurat(tmp_pxl_file)
+#'
+#' # Now we can load graphs
+#' seur_obj <- LoadCellGraphs(seur_obj, cells = colnames(seur_obj)[1])
+#'
+#' # Removing or moving PXL file will make graphs inaccessible
+#' fs::file_delete(tmp_pxl_file)
+#' \dontrun{
+#' # This will now fail since the PXL file is missing
+#' seur_obj <- LoadCellGraphs(seur_obj, force = TRUE)
+#' }
+#'
+#' # We can restore paths to the PXL file using RestorePaths.
+#' # All we need is to provide the path to a directory where the PXL file is located.
+#' pxl_files_dir <- system.file("extdata/five_cells",
+#'                             package = "pixelatorR")
+#' seur_obj <- RestorePaths(seur_obj, pxl_files_dir = pxl_files_dir)
+#'
+#' # Now LoadCellGraphs should work as expected
+#' seur_obj <- LoadCellGraphs(seur_obj, force = TRUE)
+#'
+#' @export
+#'
+RestorePaths.MPXAssay <- function (
+  object,
+  pxl_files_dir,
+  verbose = TRUE,
+  ...
+) {
+
+  # Check that pxl_files_dir exists
+  if (!fs::file_exists(pxl_files_dir)) {
```

Comment:
```suggestion
  if (!fs::dir_exists(pxl_files_dir)) {
```

Reply:
Just in case the user would supply a file name instead of a dir.

Reply:
Good catch!